climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+        The set of impact functions.
+    haz : Hazard
+        The hazard instance.
+    agg : string, optional
+        If 'agg', the impact is averaged over all points in each geometry.
+        If 'sum', the impact is summed over all points in each geometry.
+        The default is 'sum'.
+
+    Returns
+    -------
+    Impact
+        Impact object with the impact per geometry.
+
+    """
+    # compute impact
+    impact_pnt = Impact()
+    impact_pnt.calc(exp_pnt, impf_set, haz, save_mat=True)
+
+    return impact_pnt_agg(impact_pnt, exp_pnt, agg)
+
+
+def impact_pnt_agg(impact_pnt, exp_pnt, agg):
+    """
+    Aggregate the impact per geometry.
+
+    The output Impact object contains an extra attribute 'geom_exp'
+    containing the geometries.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per exposure point (lines of exp_pnt)
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first for geometries,
+        second for the point disaggregation of the geometries.
+    agg : string, optional
+        If 'agg', the impact is averaged over all points in each geometry.
+        If 'sum', the impact is summed over all points in each geometry.
+        The default is 'sum'.
+
+    Returns
+    -------
+    Impact
+        Impact object with the impact per geometry.
+
+    """
+
+    # aggregate impact
+    mat_agg = aggregate_impact_mat(impact_pnt, exp_pnt.gdf, agg)
+
+    #Write to impact obj
+    impact_agg = set_imp_mat(impact_pnt, mat_agg)
+
+    #Add exposure representation points as coordinates
+    repr_pnts = gpd.GeoSeries(
+        exp_pnt.gdf['geometry_orig'][:,0].apply(
+            lambda x: x.representative_point()
+            )
+        )
+    impact_agg.coord_exp = np.array([repr_pnts.y, repr_pnts.x]).transpose()
+    #Add geometries
+    impact_agg.geom_exp = exp_pnt.gdf.xs(0, level=1)\
+        .set_geometry('geometry_orig')\
+            .geometry.rename('geometry')
+
+    return impact_agg
+
+def aggregate_impact_mat(imp_pnt, gdf_pnt, agg):
+    """
+    Aggregate impact matrix given geodataframe or disaggregated polygons.
+
+    Parameters
+    ----------
+    impact_pnt : Impact
+        Impact object with impact per point (lines of gdf_pnt)
+    gdf_pnt : GeoDataFrame
+        Exposures geodataframe with a double index, first for polygon geometries,
+        second for the point disaggregation of the polygons.
+    agg : string
+        If 'agg', the impact is averaged over all points in each polygon.
+        If 'sum', the impact is summed over all points in each polygon.
+
+    Returns
+    -------
+    sparse.csr_matrix
+        matrix num_events x num_polygons with impacts.
+
+    """
+    # aggregate impact
+    mi = gdf_pnt.index
+    col_geom = mi.get_level_values(level=0)
+    def convert_str_idx_to_int(idx):
+        """Converts string multi-index level 0 to integer index"""
+        return np.sort(np.unique(idx, return_inverse=True)[1])
```

Comment:
Why is it necessary to define this in here? Or just to "hide it away" because it's a util func of a util func?

Reply:
I found this a tad bit more clear to read... but yes, one could also do with a comment.